### DISCUSSION TITLE
2 Previews of the same thing not equivalent

### DISCUSSION THREAD

**Discussion by TuanaCelik** [vercel/vercel#7812] [Help]
Hi, I'm quite new to next and vercel so this might be something simple I'm overlooking but I'm having trouble understanding how this behaviour occurs and what might be the cause of it. Any help would be greatly appreciated!

I am using vercel with my next.js website. And when there's a PR vercel creates a preview deployment which you can access from the PR as so:

<img width="918" alt="image" src="https://user-images.githubusercontent.com/15802862/168785706-92e58d23-3bb9-4aaa-b38b-db49e2f4eeb6.png">

For example, the preview url here includes the name of my branch: https://haystack-website-git-tuana-tutorials-deepset-overnice.vercel.app/

On vercel though, you can go to that same preview deployment and there's a 'Visit' button as so:

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/15802862/168786145-cf10ae86-5d7b-4ab6-85f9-93df9288a618.png">

Here you can see the domain is different, so you go to the following URL: https://haystack-website-5t0najmtn-deepset-overnice.vercel.app/

My issue here is that the resulting previews aren't actually equivalent. For example, int the first URL I get some 404s, but in the other I don't.

The reason I'm asking this is because I'm finding it difficult to hunt down why we randomly get some 404s, which don't even appear in the build logs. Then, when I see 2 preview URLs for the same deployment, one of which that have the 404s and the other that doesn't it's even more confusing. Any help understanding what the difference of these 2 previews are and why the behaviour is different would be much appreciated.

**Comment by leerob** [comment]
Hey!

This is the difference between **branch URLs** and **deployment URLs**, or a specific commit: https://vercel.com/docs/concepts/deployments/automatic-urls

Could you share a video/more details about the 404s? That sounds unexpected.
